src/unistd.rs

```diff
@@ -375,16 +377,23 @@ pub fn sleep(seconds: libc::c_uint) -> c_uint {
 }
 
 #[inline]
-pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<RawFd> {
-    let res = try!(template.with_nix_path(|path| {
-        let mut path_copy = path.to_bytes_with_nul().to_owned();
-        let c_template: *mut c_char = path_copy.as_mut_ptr() as *mut c_char;
+pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
```

Comment:
I don't like this too much because it forces an allocation and copy. See https://github.com/nix-rust/nix/pull/365#issuecomment-216374309 for another option.